### PR TITLE
Run Enso Cloud tests on the CI

### LINKED ISSUE #9523: Run Enso Cloud tests on the CI
Currently the Enso Cloud tests are only being run locally, because they require an account set up.

To run these tests one has to set the following variables:
```
ENSO_CLOUD_API_URI="https://7aqkn3tnbc.execute-api.eu-west-1.amazonaws.com/"
ENSO_CLOUD_CREDENTIALS_FILE="~\.enso\credentials"
ENSO_RUN_REAL_CLOUD_TEST=1
```

- [x] We should create an account on the Enso Cloud that can be used by the CI to run these tests. 
- [x] We need then to authenticate this account on the CI and generate the `credentials` file (like the one generated by the GUI).
- [x] We need to ensure that the account is set up for the tests.
	- If done before #9291 we need to manually add some starting files that are expected in the tests.
	- If done after #9291, we could set these files as part of the tests.
	- Any write tests need to run in an independent sub-directory, because there can be multiple CI instances running these tests on the same account in parallel, and they should not interfere with each other.

Ideally, these tests should be run as part of our nightly/on-demand "Integration Test Suite". But before it is introduced, we will probably just run them on the main CI workflow, as long as it's not too flaky.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Closes #9523 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "rust-analyzer.linkedProjects": [
-    "./app/gui2/rust-ffi/Cargo.toml"
```
Comment (radeusgd) [inline .vscode/settings.json:3]: These fixes in `settings.json` were needed to be able to use the Rust extensions of VS Code. Without changes it was crashing upon project import.

File: build/build/src/enso.rs [hunk 2]
```diff
@@ -160,6 +172,8 @@ impl BuiltEnso {
                 only.iter().any(|test| test.contains("Microsoft_Tests")),
         };
 
+        let cloud_credentials_file = cloud_tests::prepare_credentials_file().await?;
```
Comment (radeusgd) [inline build/build/src/enso.rs]: It seems that this temp file does not live long enough to be seen by running tests.

e.g. I can see that the path is set in env variables, but while the tests are running the file does not exist.

I assume it is successfully created, so it is probably dropped too soon.

@Frizi do you have any hints maybe what I can do to extend its lifetime until the `self.run_test` below finishes?

Comment (Frizi) [reply build/build/src/enso.rs]: Yes it is dropped at the end of the function, once the run future is returned, i.e. before the execution begins. You have to manually drop it only after the program has finished running (i.e. inside the future, past the await point).

```rs
async move {
    let result = command?.run_ok().await;
    drop(cloud_credentials_file);
    result
}
```

Comment (radeusgd) [reply build/build/src/enso.rs]: Hmm, but I'm running several commands in there (it's inside `into_iter().map`), so I'm not sure if that will work?

Comment (Frizi) [reply build/build/src/enso.rs]: Right, I missed that. In this case you have to `drop` it after the `future::join_all` call, which is the next await point after the loop generating invididual futures.

**Review by Frizi — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]

File: test/Base_Tests/src/System/File_Spec.enso [hunk 3]
```diff
@@ -366,12 +366,12 @@ add_specs suite_builder =
...
-        is_correct_working_directory = (File.current_directory . normalize . path) == current_project_root.absolute.normalize.path
+        is_correct_working_directory = File.current_directory.absolute.normalize.path == base_directory.absolute.normalize.path
         group_builder.specify "will resolve relative paths relative to the currently running project" pending=(if is_correct_working_directory.not then "The working directory is not set-up as expected, so this test cannot run. Please run the tests using `ensoup` to ensure the working directory is correct.") <|
-            root = File.new "."
-            root.should_be_a File
+            dot = File.new "."
+            dot.should_be_a File
             # The `.` path should resolve to the base path
-            root.absolute.normalize.path . should_equal base_directory.absolute.normalize.path
+            dot.absolute.normalize.path . should_equal base_directory.absolute.normalize.path
```
Comment (radeusgd) [inline test/Base_Tests/src/System/File_Spec.enso:374]: This test setup check was wrong. I noticed it when testing, so decided to fix it.

**Pull request merged**